<?xml version="1.0"?>

<launch>
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="model" default="$(find hector_quadrotor_description)/urdf/quadrotor.gazebo.xacro" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find multi_robot)/worlds/flood.world"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

  <group ns="drone01">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone01"/>
      <arg name="tf_prefix" value="drone01"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="0.0"/>
      <arg name="y" value="1.5"/>
    </include>
  </group>

  <group ns="drone02">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone02"/>
      <arg name="tf_prefix" value="drone02"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="1.5"/>
      <arg name="y" value="0"/>
    </include>
  </group>

  <group ns="drone03">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone03"/>
      <arg name="tf_prefix" value="drone03"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="0"/>
      <arg name="y" value="-1.5"/>
    </include>
  </group>

  <group ns="drone04">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone04"/>
      <arg name="tf_prefix" value="drone04"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="-1.5"/>
      <arg name="y" value="0"/>
    </include>
  </group>

  <group ns="drone05">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone05"/>
      <arg name="tf_prefix" value="drone05"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="1.5"/>
      <arg name="y" value="1.5"/>
    </include>
  </group>

  <group ns="drone06">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone06"/>
      <arg name="tf_prefix" value="drone06"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="1.5"/>
      <arg name="y" value="-1.5"/>
    </include>
  </group>

  <group ns="drone07">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone07"/>
      <arg name="tf_prefix" value="drone07"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="-1.5"/>
      <arg name="y" value="-1.5"/>
    </include>
  </group>

  <group ns="drone08">
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
      <arg name="name" value="drone08"/>
      <arg name="tf_prefix" value="drone08"/>
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_downward_cam.gazebo.xacro"/>
      <arg name="x" value="-1.5"/>
      <arg name="y" value="1.5"/>
    </include>
  </group>


</launch>
